### PR TITLE
Consistency check mandatory properties

### PR DESCRIPTION
**Pull request by thobe**

### CODE REVIEW

File: community/consistency-check/src/main/java/org/neo4j/consistency/checking/SchemaRecordCheck.java [hunk 1]
```diff
@@ -169,6 +173,16 @@ else if ( phase == Phase.CHECK_OBLIGATIONS )
         }
     }
 
+    private void checkMandatoryPropertyConstraintRule( MandatoryPropertyConstraintRule rule,
+            CheckerEngine<DynamicRecord,ConsistencyReport.SchemaConsistencyReport> engine, DynamicRecord record,
+            RecordAccess records )
+    {
+        if ( phase == Phase.CHECK_RULES )
+        {
+            // TODO: implement this
```
Comment (pontusmelke) [inline community/consistency-check/src/main/java/org/neo4j/consistency/checking/SchemaRecordCheck.java]: Does this need to happen before merging?

Comment (thobe) [inline community/consistency-check/src/main/java/org/neo4j/consistency/checking/SchemaRecordCheck.java]: Ah, right, yes. perhaps it should. good catch.

File: community/consistency-check/src/main/java/org/neo4j/consistency/checking/full/MandatoryPropertyChecker.java [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+        }
+
+        @Override
+        PrimitiveIntSet mandatoryPropertiesFor( NodeRecord record,
+                CheckerEngine<NodeRecord,ConsistencyReport.NodeConsistencyReport> engine, RecordAccess records )
+        {
+            PrimitiveIntSet keys = null;
+            for ( Long label : getListOfLabels( record, records, engine ) )
+            {
+                int[] labelKeys = mandatory.get( label.intValue() );
```
Comment (lutovich) [inline community/consistency-check/src/main/java/org/neo4j/consistency/checking/full/MandatoryPropertyChecker.java]: I think this variable should be named `propertyKeys`. Same for one in `RelationshipChecker#mandatoryPropertiesFor`

File: community/consistency-check/src/main/java/org/neo4j/consistency/checking/full/MandatoryPropertyChecker.java [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+public class MandatoryPropertyChecker extends CheckDecorator.Adapter
+{
+    private final PrimitiveIntObjectMap<int[]> nodes = Primitive.intObjectMap();
+    private final PrimitiveIntObjectMap<int[]> relationships = Primitive.intObjectMap();
+
+    public MandatoryPropertyChecker( RecordStore<DynamicRecord> schemaStore )
+    {
+        for ( Iterator<MandatoryPropertyConstraintRule> rules =
+              new SchemaStorage( schemaStore ).schemaRules( MandatoryPropertyConstraintRule.class );
+              safeHasNext( rules ); )
```
Comment (lutovich) [inline community/consistency-check/src/main/java/org/neo4j/consistency/checking/full/MandatoryPropertyChecker.java:60]: Why `rules.hasNext()` is unsafe?

Comment (thobe) [inline community/consistency-check/src/main/java/org/neo4j/consistency/checking/full/MandatoryPropertyChecker.java:60]: Look at what `safeHasNext` does...

If the schemaStore is inconsistent, `hasNext` will throw an exception when trying to read the inconsistent record. `safeHasNext` allows us to just continue to the next record at that point.

**Pull request merged**